code/modules/reagents/chemistry/holder.dm

```diff
@@ -614,12 +614,20 @@
 	for(var/_reagent in cached_reagents)
 		var/datum/reagent/R = _reagent
 		if(R.type == reagent)
-			if(my_atom && isliving(my_atom))
-				var/mob/living/M = my_atom
+			var/mob/living/mob_consumer
+
+			if (isliving(my_atom))
+				mob_consumer = my_atom
+			else if (istype(my_atom, /obj/item/organ/stomach))
+				var/obj/item/organ/stomach/belly = my_atom
+				mob_consumer = belly.owner
```

Comment:
I hate this. Let me know if there's anything better.

Reply:
holder should just proc/get_holder_mob() and return the mob for any organ or container like this
and the check could be for all organs not just stomach as i think of it.

Reply:
any reason why this is specific to stomachs?

Reply:
I can make it check just for organs, if that typecheck still seems reasonable to you.

Reply:
this is fine as is, i am just overthinking it and ensuring no organ has a reagent that could be acting on you